### PR TITLE
feat: adding testcases for reserve exactly or exceeding global net de…

### PR DESCRIPTION
**Pull request by davidprepo**
Add 2 testcases for ManagerWithdrawHook to ensure that hook does not revert when getReserve is exactly or exceeds getGlobalNetDepositAmount

### CODE REVIEW

File: apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts [hunk 1]
```diff
@@ -374,5 +374,21 @@ describe('=> ManagerWithdrawHook', () => {
 
       await managerWithdrawHook.connect(user).hook(user.address, IGNORED_ARGUMENT, amountToWithdraw)
     })
+
+    it('allows withdrawal when reserve is higher than global net deposit amount', async () => {
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts:378]: Should use "if" wording and more concise wording for comparisons, so `allows withdrawal if reserve is > global net deposit amount`. Please do the same for the testcase below.

File: apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts [hunk 2]
```diff
@@ -374,5 +374,21 @@ describe('=> ManagerWithdrawHook', () => {
...
+      depositRecord.getGlobalNetDepositAmount.returns(TEST_GLOBAL_DEPOSIT_CAP)
+      collateral.getReserve.returns(TEST_GLOBAL_DEPOSIT_CAP.add(1))
+
+      const amountToWithdraw = 1
+      await managerWithdrawHook.connect(user).hook(user.address, IGNORED_ARGUMENT, amountToWithdraw)
+    })
+
+    it('allows withdrawal when reserve is exactly at global net deposit amount', async () => {
```
Comment (ramenforbreakfast) [inline apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts:386]: I would put this above the previous testcase.

**Review by ramenforbreakfast — APPROVED** [review]
LGTM besides some minor nits

File: apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts (hunk 1 above)
Comment (XavierEkkel) [reply apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts:378]: `is` is also a filler word we want to avoid, so this would be better:
`allows withdrawal if reserve > global net deposit amount`

Comment (davidprepo) [reply apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts:378]: Understood, will follow in future.

File: apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts (hunk 2 above)
Comment (davidprepo) [reply apps/smart-contracts/core/test/ManagerWithdrawHook.test.ts:386]: rgr
